components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -22,37 +29,77 @@
 #include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-shared.h"
 #include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
 #include "brave/components/ai_chat/core/common/pref_names.h"
+#include "components/os_crypt/async/browser/os_crypt_async.h"
 #include "components/prefs/pref_service.h"
 
 namespace ai_chat {
 namespace {
 
+constexpr base::FilePath::StringPieceType kDBFileName =
+    FILE_PATH_LITERAL("AIChat");
+
 static const auto kAllowedSchemes = base::MakeFixedFlatSet<std::string_view>(
     {url::kHttpsScheme, url::kHttpScheme, url::kFileScheme, url::kDataScheme});
 
+std::vector<mojom::Conversation*> FilterVisibleConversations(
+    std::map<std::string, mojom::ConversationPtr>& conversations_map) {
+  std::vector<mojom::Conversation*> conversations;
+  for (const auto& kv : conversations_map) {
+    auto& conversation = kv.second;
+    // Conversations are only visible if they have content
+    if (!conversation->has_content) {
+      continue;
+    }
+    conversations.push_back(conversation.get());
+  }
```

Comment:
In what scenarios are there not content? Is this so we don't show things in the sidebar until there's something there?

Just want to confirm we don't save empty conversations

Reply:
New in-memory conversations don't have content. Conversations begin in-memory and are only persisted when they get new conversation entries - see `OnConversationEntryAdded`.

Looking at this again, I suppose we could prevent having to have this field by checking if the conversation has an active ConversationHandler and if that handler reports it has any conversation entries. This property is just an optimization preventing looking that up each time, but I think it's worthwhile given `GetVisibleConversations` and `FilterVisibleConversations` gets called often.